nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/service/StandardControllerServiceNode.java

```diff
@@ -121,6 +122,11 @@ public Resource getResource() {
     }
 
     @Override
+    public boolean isRestricted() {
```

Comment:
This cannot be moved as the logic to get the underlying class is not consistent between Processor's, Controller Service's, and Reporting Task's. Alternatively, we could introduce another method that is responsible for returning the underlying instance type. This would allow us to move `isRestricted` up but I didn't think it was necessary.